patterns/terraform/main.tf

```diff
@@ -59,30 +58,51 @@ locals {
 # EKS Cluster
 ################################################################################
 
-#tfsec:ignore:aws-eks-enable-control-plane-logging
 module "eks" {
   source  = "terraform-aws-modules/eks/aws"
-  version = "~> 19.13"
+  version = "~> 20.20"
 
-  cluster_name                   = local.name
-  cluster_version                = "1.27"
-  cluster_endpoint_public_access = true
+  cluster_name    = local.name
+  cluster_version = "1.30"
+
+  cluster_endpoint_private_access = true
+  cluster_endpoint_public_access  = true
+  kms_key_enable_default_policy   = true
+
+  # Give the Terraform identity admin access to the cluster
+  # which will allow resources to be deployed into the cluster
+  enable_cluster_creator_admin_permissions = true
+
+  # EKS Addons
+  cluster_addons = {
+    coredns = {
+      most_recent = true
+    }
+    kube-proxy = {
+      most_recent = true
+    }
+    vpc-cni = {
+      before_compute = true # Ensure the addon is configured before compute resources are created
+      most_recent    = true
+    }
+  }
 
   vpc_id     = module.vpc.vpc_id
   subnet_ids = module.vpc.private_subnets
 
-  manage_aws_auth_configmap = true
-
   eks_managed_node_groups = {
     initial = {
-      instance_types = ["m5.xlarge"]
-      max_size       = 3
-      min_size       = 3
-      desired_size   = 3
+      instance_types = ["m5.large"]
+
+      min_size     = 1
+      max_size     = 5
+      desired_size = 3
     }
   }
 
   tags = local.tags
+
+  depends_on = [module.vpc]
```

Comment:
Not needed as there is an implicit dependency when you provide vpc id and subnets.